pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -40,110 +51,193 @@ func (r *Condition) Update(other Condition) {
 	r.Type = other.Type
 	r.Status = other.Status
 	r.Reason = other.Reason
+	r.Category = other.Category
 	r.Message = other.Message
 	r.LastTransitionTime = metav1.NewTime(time.Now())
 }
 
 func (r Condition) Equal(other Condition) bool {
 	return r.Type == other.Type &&
 		r.Status == other.Status &&
+		r.Category == other.Category &&
 		r.Reason == other.Reason &&
 		r.Message == other.Message
 }
 
 // Managed collection of conditions.
 // Intended to be included in resource Status.
+// List - The list of conditions.
+// staged - In `staged` mode, the search methods like
+//          HasCondition() filter out un-staged conditions.
+// -------------------
+// Example:
+//
+// thing.Status.BeginStagedConditions()
+// thing.Status.SetCondition(c)
+// thing.Status.SetCondition(c)
+// thing.Status.SetCondition(c)
+// thing.Status.EndStagedConditions()
+// thing.Status.SetReady(
+//     !thing.Status.HasBlockerCondition(),
+//     "Resource Ready.")
+//
 type Conditions struct {
-	Conditions []Condition `json:"conditions"`
+	List   []Condition `json:"conditions"`
+	staged bool
+}
+
+// Begin staged conditions.
+func (r *Conditions) BeginStagedConditions() {
+	r.staged = true
+	if r.List == nil {
+		return
+	}
+	for index := range r.List {
+		condition := &r.List[index]
+		condition.staged = false
+	}
+}
+
+// End staged conditions.
+func (r *Conditions) EndStagedConditions() {
+	r.staged = false
```

Comment:
This sets an entire group of conditions as `staged = false`.  Not sure what that means.

Reply:
See above, wonder if `staging` would make this clearer if it's doing what I think it is.

Reply:
No, This puts the `Conditions` collection in _non-staging_ mode and deletes any of the _un-staged_ conditions (Condition.staged=false).